DAOS-11601 object: using DTX for EC agg peer update

liuxuezhao opened the pull request:
Required-githooks: true

TS/MVCC conflict issue not resolved yet.

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

liuxuezhao committed:
DAOS-11601 object: using DTX for EC agg peer update

Required-githooks: true

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

daosbuild1 commented on src/object/cli_ec.c:
```diff
@@ -1684,22 +1684,83 @@ obj_ec_encode(struct obj_reasb_req *reasb_req)
+			siod->siod_off = 0;
+		}
+		EC_TRACE("i %d tgt %u idx %u nr %u, start "DF_U64"\n",
+			i, siod->siod_tgt_idx, siod->siod_idx,
```
```suggestion
			 i, siod->siod_tgt_idx, siod->siod_idx,
```

daosbuild1 commented on src/object/srv_obj.c:
```diff
@@ -3973,15 +3975,27 @@ ds_cpd_handle_one(crt_rpc_t *rpc, struct daos_cpd_sub_head *dcsh,
-				&iohs[i], dth);
+		if (likely(!ec_agg_remove)) {
+			rc = vos_update_begin(ioc->ioc_vos_coh,
+					dcsr->dcsr_oid, dcsh->dcsh_epoch.oe_value,
```
```suggestion
					      dcsr->dcsr_oid, dcsh->dcsh_epoch.oe_value,
```

daosbuild1 commented on src/vos/vos_io.c:
```diff
@@ -2817,12 +2817,44 @@ vos_obj_update(daos_handle_t coh, daos_unit_oid_t oid, daos_epoch_t epoch,
+	int rc;
+
+	rc = vos_update_begin(coh, oid, epr->epr_hi, VOS_OF_REMOVE, (daos_key_t *)dkey,
+			       1, iod, NULL, 0, ioh, dth);
```
```suggestion
			      1, iod, NULL, 0, ioh, dth);
```

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10488/1/execution/node/145/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10488/3/execution/node/1092/log

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10488/3/execution/node/1182/log

liuxuezhao committed:
DAOS-11601 object: remove the retry of parity epoch mismatch

The obj_ec_parity_lists_match() check and retry was added to resolve the
case of EC agg's parity update inconsistency between different parity
shards. As EC agg changed to use DTX that resolved the consistency
problem, now that retry can be removed.

Required-githooks: true

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

liuxuezhao committed:
DAOS-11601 object: change dtx_leader_end

dtx_leader_end() get "coh->sch_cont" from "ioc->ioc_coh" that possibly
be NULL for rebuild container which is used for rebuild and EC
aggregation. This patch add "struct ds_cont_child *" parameter to
dtx_leader_end().

Required-githooks: true

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

liuxuezhao committed:
DAOS-11601 object: bug fix

Required-githooks: true

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10488/10/execution/node/1131/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-10488/11/execution/node/1083/log
